router/io-spec-lemmas.gobra

```diff
@@ -73,6 +73,26 @@ func absPktFutureLemma(raw []byte) {
 	assert len(pkt.CurrSeg.Future) > 0
 }
 
+ghost
+requires acc(sl.Bytes(raw, 0, len(raw)), R56)
+requires slayers.ValidPktMetaHdr(raw)
+requires len(absPkt(raw).CurrSeg.Future) > 0
+requires acc(&p.hopField, R55)
+requires p.EqAbsHopField(absPkt(raw))
+ensures  acc(&p.hopField, R55)
+ensures  acc(sl.Bytes(raw, 0, len(raw)), R56)
+ensures  slayers.ValidPktMetaHdr(raw)
+ensures  len(absPkt(raw).CurrSeg.Future) > 0
+ensures  absPkt(raw) ==
+	absPkt(raw).UpdateHopField(p.hopField.ToIO_HF())
+decreases
+func (p* scionPacketProcessor) absPktUpdateLemma(raw []byte) {
```

Comment:
I wonder if this property is stated at the most adequate level. In particular, I wonder if the most fundamental lemma here is `forall seg, hf . seg.UpdateCurr(seg.Future[0]) == seg`